addons/mrp/models/stock_rule.py

```diff
@@ -198,7 +198,9 @@ def _get_lead_days(self, product, **values):
         if not manufacture_rule:
             return delay, delay_description
         manufacture_rule.ensure_one()
-        bom = values.get('bom') or (product.variant_bom_ids or product.bom_ids)[:1]
+        company_ids = self.mapped('company_id').ids
+        product_template_bom_ids = self.env['mrp.bom'].search([('company_id', 'in', company_ids), ('product_tmpl_id', '=', product.product_tmpl_id.id)])
+        bom = values.get('bom') or (product.variant_bom_ids or product_template_bom_ids)[:1]
```

Comment:
You could just take the manufacture rule to find the BoM. Also use _bom_find with the company (it will avoid to pick BoM for another variant)